roles/import_image_and_run_aws_instance/tasks/main.yml

```diff
@@ -0,0 +1,100 @@
+        name: "{{ import_image_and_run_aws_instance_bucket_name }}"
+      register: import_image_and_run_aws_instance__bucket_info
+
+    - name: Fail when S3 bucket name does not exist
+      ansible.builtin.fail:
+        msg: "The on prem VM {{ import_image_and_run_aws_instance_bucket_name }} does not exist."
+      when: import_image_and_run_aws_instance__bucket_info.buckets | length == 0
+
+    - name: Set 'import_image_and_run_aws_instance__s3_object_key' variable
+      ansible.builtin.set_fact:
+        import_image_and_run_aws_instance__s3_object_key: "{{ import_image_and_run_aws_instance_bucket_name }}/{{ import_image_and_run_aws_instance_instance_name }}.raw"
+
+    - name: Upload raw image to S3 bucket
+      amazon.aws.s3_object:
+        bucket: "{{ import_image_and_run_aws_instance_bucket_name }}"
+        object: "{{ import_image_and_run_aws_instance__s3_object_key }}"
+        src: "{{ import_image_and_run_aws_instance_image_path }}"
+        mode: put
+
+    - name: Import image
+      amazon.aws.ec2_import_image:
+        state: present
+        task_name: "{{ import_image_and_run_aws_instance_import_image_task_name }}"
+        disk_containers:
+          - format: raw
+            user_bucket:
+              s3_bucket: "{{ import_image_and_run_aws_instance_bucket_name }}"
+              s3_key: "{{ import_image_and_run_aws_instance__s3_object_key }}"
+      register: import_image_and_run_aws_instance__import_result
+
+    # image_id and snapshot_id are not available until the import image task is completed
+    - name: Check status of the import image task
+      amazon.aws.ec2_import_image_info:
+        filters:
+          - Name: "tag:Name"
+            Values: ["{{ import_image_and_run_aws_instance_import_image_task_name }}"]
+          - Name: "task-state"
+            Values: ["completed", "active"]
+      register: import_image_and_run_aws_instance__import_image_info
+      until: import_image_and_run_aws_instance__import_image_info.import_image[0].status == "completed"
+      delay: 3
+      retries: 30
+
+    - name: Set 'import_image_and_run_aws_instance__ami_id' and 'import_image_and_run_aws_instance__snapshot_id'
+      ansible.builtin.set_fact:
+        import_image_and_run_aws_instance__ami_id: "{{ import_image_and_run_aws_instance__import_image_info.import_image[0].image_id }}"
+        import_image_and_run_aws_instance__snapshot_id: "{{ import_image_and_run_aws_instance__import_image_info.import_image[0].snapshot_details[0].snapshot_id }}"
+
+    - name: Check if an instance with the specified name alrerady exists
+      amazon.aws.ec2_instance_info:
+        filters:
+          "tag:Name": "{{ import_image_and_run_aws_instance_instance_name }}"
+          instance-state-name: [ "running", "pending", "shutting-down", "stopping", "stopped"]
+      register: import_image_and_run_aws_instance__ec2_instance_info
+
+    - name: Fail when an instance with the specified name already exists
```

Comment:
I wonder if we want to run this check and failure earlier in the role? Since it doesn't look like it depends on any of the previous steps and this is really checking whether the user input for the `import_image_and_run_aws_instance_instance_name` variable will be valid, maybe we give the user that info before the role starts doing other things?

Reply:
Moved at the beginning!